Tests/SonarScanner.MSBuild.Test/AnalysisWarning/WarningsSerializerTest.cs

```diff
@@ -0,0 +1,69 @@
+﻿/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2022 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarScanner.MSBuild.AnalysisWarning;
+
+namespace SonarScanner.MSBuild.Test.AnalysisWarning
+{
+    [TestClass]
+    public class WarningsSerializerTest
+    {
+        public TestContext TestContext { get; set; }
+
+        [TestMethod]
+        public void SerializeToFile_InvalidWarningsCollections_Throws()
+        {
+            Action act = () => WarningsSerializer.Serialize(null, "filePath");
+            act.Should().Throw<ArgumentNullException>().WithMessage(@"Value cannot be null.
+Parameter name: warnings");
+        }
```

Comment:
The scaffolding around this can be simplified to a single line arrow method to remove boring test. See
https://github.com/SonarSource/sonar-dotnet/blob/071df8117ddc6d639e04aac5f3f9c9932261f3fc/analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/SymbolicCheckListTest.cs#L31-L32

The underlying idea is that as long as the parameter name is correct (the only input we provide), we should be fine. The message is auto-generated and differs for different SDKs

Reply:
I think Pavel's idea was to also simplify `to a single line arrow method to remove boring test`